pkg/vhdbuilder/datamodel/component_configs.go

```diff
@@ -134,29 +149,13 @@ func (k *KubeProxyImages) ToImageList() []string {
 
 	if k.DockerKubeProxyImages != nil && k.DockerKubeProxyImages.ContainerImages != nil {
 		for _, image := range k.DockerKubeProxyImages.ContainerImages {
-			if image.Amd64OnlyVersions != nil {
-				amd64OnlyImageList, _ := toImageList(image.DownloadURL, image.Amd64OnlyVersions)
-				ret = append(ret, amd64OnlyImageList...)
-			}
-
-			if image.MultiArchVersions != nil {
-				multiArchImageList, _ := toImageList(image.DownloadURL, image.MultiArchVersions)
-				ret = append(ret, multiArchImageList...)
-			}
+			ret = processProxyImages(image, ret)
 		}
 	}
 
 	if k.ContainerdKubeProxyImages != nil && k.ContainerdKubeProxyImages.ContainerImages != nil {
 		for _, image := range k.ContainerdKubeProxyImages.ContainerImages {
-			if image.Amd64OnlyVersions != nil {
-				amd64OnlyImageList, _ := toImageList(image.DownloadURL, image.Amd64OnlyVersions)
-				ret = append(ret, amd64OnlyImageList...)
-			}
-
-			if image.MultiArchVersions != nil {
-				multiArchImageList, _ := toImageList(image.DownloadURL, image.MultiArchVersions)
-				ret = append(ret, multiArchImageList...)
-			}
+			ret = processProxyImages(image, ret)
```

Comment:
might make more sense to have `processProxyImages` operate on a *[]string and have it be a void func instead of returning the updated slice every iteration